packages/renderers/renderer-preact/server.js

```diff
@@ -10,7 +10,15 @@ function check(Component, props, children) {
   }
 
   const { html } = renderToStaticMarkup(Component, props, children);
-  return typeof html === 'string';
+
+  if (typeof html !== 'string') {
+    return false;
+  }
+
+  // There are edge cases (SolidJS) where Preact *might* render a string,
+  // but components would be <undefined></undefined>
+
+  return !/\<undefined\>/.test(html);
```

Comment:
Solid components can run through Preact's `renderToString` fine! But they return `<undefined>`... So that's a weird case to check for here.

Reply:
very strange.

Can this idea of "check()" now go away? It sounds like we have added the ability to know which framework a file was transformed for. If we could surface that info somehow, then we wouldn't need this runtime check...

I'm okay if this is a follow up PR, but maybe its actually really easy?

Reply:
I'll have to think about this! The way it's structured right now, `check` is a runtime concern, whereas this compilation is a build-time concern. 

There might be a clean way to refactor this, but I think that's out of scope for this PR.